File: src/main/java/net/gtaun/shoebill/common/vehicle/VehicleLifecycleObject.java
package net.gtaun.shoebill.common.vehicle;

import net.gtaun.shoebill.common.AbstractShoebillContext;
import net.gtaun.shoebill.object.Vehicle;
import net.gtaun.util.event.EventManager;

/**
 * Created by marvin on 01.11.14.
 */
public abstract class VehicleLifecycleObject extends AbstractShoebillContext
{
    protected final Vehicle vehicle;

    public VehicleLifecycleObject(EventManager eventManager, Vehicle vehicle)
    {
        super(eventManager);
        this.vehicle = vehicle;
    }

    public Vehicle getVehicle()
    {
        return vehicle;
    }
}

File: src/main/java/net/gtaun/shoebill/common/vehicle/VehicleLifecycleObject.kt
package net.gtaun.shoebill.common.vehicle

import net.gtaun.shoebill.Shoebill
import net.gtaun.shoebill.common.LifecycleObject
import net.gtaun.shoebill.entities.Vehicle

abstract class VehicleLifecycleObject(val vehicle: Vehicle) : LifecycleObject(Shoebill.get().eventManager)
